### PR TITLE
Add DeprecateImages step type, modify AddDependency to take *Step instead of string

### PR DESCRIPTION
**Pull request by adjackura**

### CODE REVIEW

File: daisy/disk_test.go [hunk 1]
```diff
@@ -30,9 +30,9 @@ func TestDiskRegisterAttachment(t *testing.T) {
 	s2, _ := w.NewStep("s2")
 	att, _ := w.NewStep("att")
 	det, _ := w.NewStep("det")
-	w.AddDependency("det", "att")
-	w.AddDependency("s", "det")
-	w.AddDependency("iw2Step", "iwStep")
+	w.AddDependency(det, att)
```
Comment (crunk1) [inline daisy/disk_test.go]: I like this better than what I had.

File: daisy/compute/compute.go [hunk 2]
```diff
@@ -267,6 +268,16 @@ func (c *client) DeleteInstance(project, zone, name string) error {
 	return c.i.operationsWait(project, zone, op.Name)
 }
 
+// DeprecateImage sets deprecation status on a GCE image.
+func (c *client) DeprecateImage(project, name string, deprecationstatus *compute.DeprecationStatus) error {
```
Comment (crunk1) [inline daisy/compute/compute.go]: Is it getting annoying having to wrap each API method? If so, we can talk about how to better approach the compute API.

File: daisy/step_deprecate_images.go [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+	"sync"
+
+	"google.golang.org/api/compute/v1"
+)
+
+// DeprecateImages is a Daisy DeprecateImage workflow step.
+type DeprecateImages []*DeprecateImage
+
+// DeprecateImage sets the deprecation status on a GCE image.
+type DeprecateImage struct {
```
[the remark this answers is not in the document]
Comment (adjackura) [reply daisy/step_deprecate_images.go]: No, the deprecate call just takes a DeprecationStatus struct, which I have here.

File: daisy/step_deprecate_images.go [hunk 4]
```diff
@@ -0,0 +1,98 @@
...
+		}
+
+		if imageURLRgx.MatchString(di.Image) {
+			di.Image = extendPartialURL(di.Image, di.Project)
+		}
+	}
+	return nil
+}
+
+func (d *DeprecateImages) validate(ctx context.Context, s *Step) dErr {
```
[the remark this answers is not in the document]
Comment (adjackura) [reply daisy/step_deprecate_images.go]: yes, nice catch

**Pull request merged**